p2p/node/p2p_services.go

```diff
@@ -65,11 +61,14 @@ func (p *P2PNode) requestFromPeer(peerID peer.ID, topic *pubsubManager.Topic, re
 	if err != nil {
 		return nil, err
 	}
+
+	requestTimer := time.NewTimer(requestManager.C_requestTimeout)
+	defer requestTimer.Stop()
 	var recvdType interface{}
 	select {
 	case recvdType = <-dataChan:
 		break
-	case <-time.After(requestManager.C_requestTimeout):
+	case <-requestTimer.C:
```

Comment:
Why not keep time.After?

Reply:
yeah changing it, both should be the same but no need to change it